### PR TITLE
[schema] Add more strict CRDB index comparison

### LINKED ISSUE #4143: CRDB schema change tests not covering constraints on indices sufficiently
See https://github.com/oxidecomputer/omicron/pull/4141 for context -- I haven't dug into the root issue here, but the minimal repro case is:

dbinit.sql:
```sql
CREATE INDEX IF NOT EXISTS lookup_zpool_by_disk on omicron.public.zpool (        
    physical_disk_id                                                             
) WHERE physical_disk_id IS NOT NULL AND time_deleted IS NULL;
```

up1.sql:
```sql
CREATE INDEX IF NOT EXISTS lookup_zpool_by_disk ON omicron.public.zpool (physical_disk_id, id) WHERE time_deleted IS NULL; 
```

These are not the same, but appear to be passing the `dbinit_equals_sum_of_all_up` test within https://github.com/oxidecomputer/omicron/blob/9a1f6bc5da1efed13dabbe14ec5066002cfd0851/nexus/tests/integration_tests/schema.rs .

My suspicion: More tables from https://www.cockroachlabs.com/docs/stable/information-schema need to be compared.

### PR DESCRIPTION
**Pull request by smklein**
Adds more schema comparison for DB schemas, and adds a couple regression tests for the specific index mismatch.

Fixes #4143

### CODE REVIEW

File: nexus/tests/integration_tests/schema.rs [hunk 1]
```diff
@@ -432,6 +436,16 @@ const CHECK_CONSTRAINTS: [&'static str; 4] = [
     "check_clause",
 ];
 
+const CONSTRAINT_COLUMN_USAGE: [&'static str; 7] = [
```
Comment (smklein) [inline nexus/tests/integration_tests/schema.rs:439]: Note: This wasn't used by the regression tests, but I figured I'd add it as "yet another check to validate that the DB schemas match".

See: https://www.cockroachlabs.com/docs/stable/information-schema#constraint_column_usage

File: nexus/tests/integration_tests/schema.rs [hunk 2]
```diff
@@ -456,29 +470,50 @@ const REFERENTIAL_CONSTRAINTS: [&'static str; 8] = [
...
     "table_schema",
     "table_name",
     "non_unique",
     "index_schema",
     "index_name",
+    "seq_in_index",
     "column_name",
     "direction",
+    "storing",
+    "implicit",
```
Comment (smklein) [inline nexus/tests/integration_tests/schema.rs:484]: "implicit" was the one we ended up using for the "column comparison" regression tests, but the others also seem useful to have.

File: nexus/tests/integration_tests/schema.rs [hunk 3]
```diff
@@ -456,29 +470,50 @@ const REFERENTIAL_CONSTRAINTS: [&'static str; 8] = [
...
 ];
 
+const PG_INDEXES: [&'static str; 5] =
+    ["schemaname", "tablename", "indexname", "tablespace", "indexdef"];
```
Comment (smklein) [inline nexus/tests/integration_tests/schema.rs:488]: Note: These are not actually from "InformationSchema", but rather, from https://www.cockroachlabs.com/docs/stable/pg-catalog .

This comparison was necessary to catch the differing "WHERE" clauses.

File: nexus/tests/integration_tests/schema.rs [hunk 4]
```diff
@@ -456,29 +470,50 @@ const REFERENTIAL_CONSTRAINTS: [&'static str; 8] = [
...
+    "constraint_catalog",
+    "constraint_schema",
+    "constraint_name",
+    "table_catalog",
+    "table_schema",
+    "table_name",
+    "constraint_type",
+    "is_deferrable",
+    "initially_deferred",
+];
```
Comment (smklein) [inline nexus/tests/integration_tests/schema.rs:503]: Similar to `CONSTRAINT_COLUMN_USAGE`, this wasn't necessary here, but it seems useful to ensure that it doesn't get out-of-sync regardless.

https://www.cockroachlabs.com/docs/stable/information-schema#table_constraints

**Comment by smklein** [comment]
Added https://github.com/oxidecomputer/omicron/pull/4153 as a follow-up PR for more things that I wanted to test, but which were unrelated to the original issue.

**Review by leftwo — APPROVED** [review]
I'm not up to snuff on the precise details of how the database works, but I can follow
the flow here and for sure the specific case I had in 4143 would be caught here and
the test case confirms that.

**Pull request merged**